### DISCUSSION TITLE
Taro3.2如何在android4上运行的经验分享

### DISCUSSION THREAD

**Discussion by chenxinbin** [NervJS/taro#9493] [Q & A] [answered]
背景：
……

步骤：
1. 修改Taro项目和taro-native-shell的react-native的版本（0.63.4最低支持API16，后续版本是API21）。
`npm install react-native@0.63.4` 
2  启用分包
    文件：taro-native-shell\android\app\build.gradle
    修改：android/defaultConfig 节点下增加 multiDexEnabled true

    文件：taro-native-shell\android\app\src\main\java\com\xinditu\screen\MainApplication.java
    修改：
     1. 增加引入 import androidx.multidex.MultiDex;
     2. 在MainApplication类增加以下方发：
`@Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        MultiDex.install(this);
}`

编译后后的APK可以在安卓4上正常运行。总结起来挺简单，但解决起来也费了好长时间，遇到好几个错误，最后总结一下，防止其他人踩坑。

**Comment by chenxinbin** [comment, marked as answer]
具体背景和排错细节可以访问 简书上的文章。https://www.jianshu.com/p/8653941557eb

**Comment by tourze** [comment]
就是版本跟分包问题。
实际因为有些lib对低版本android不支持，还是难搞
